pageserver/src/import_datadir.rs

```diff
@@ -617,3 +619,27 @@ async fn read_all_bytes(reader: &mut (impl AsyncRead + Unpin)) -> Result<Bytes>
     reader.read_to_end(&mut buf).await?;
     Ok(Bytes::from(buf))
 }
+
+pub async fn create_tar_zst(pgdata_path: &Path) -> Result<Vec<u8>> {
+    let mut paths = Vec::new();
+    for entry in WalkDir::new(pgdata_path) {
+        let entry = entry?;
+        let metadata = entry.metadata().expect("error getting dir entry metadata");
+        if metadata.is_file() {
+            paths.push(entry.path().to_owned());
+        }
+    }
+    // Don't rely on file system order for listing as it may be non-deterministic
+    paths.sort();
+    let zstd = async_compression::tokio::write::ZstdEncoder::new(Vec::new());
+    let mut builder = Builder::new(zstd);
+    // Use reproducible header mode
+    builder.mode(HeaderMode::Deterministic);
+    for path in paths {
+        builder.append_path(path).await?;
```

Comment:
This looks like it could contain the absolute path to the initdb for each file. Does tokio-tar automatically relativize them to some part?

Reply:
Indeed there is even an error if you pass it absolute paths:

```
stderr: command failed: Error: Error: paths in archives must be relative when setting path for 
```

I have now pushed a version that uses `append_path_with_name`.